### PR TITLE
Some cleanup and refactoring

### PR DESCRIPTION
**Pull request by alexcope**
This PR
1. Replaces jwks.evp_pkey_ with jwks.rsa_ because the evp_pkey_ was
being used exclusively as an RSA Key.
2. Removes the PEM parse option. This option was unused and was
confusing because it expected invalid PEM as input.

### CODE REVIEW

File: src/jwks.cc [hunk 1]
```diff
@@ -329,8 +295,8 @@ Status extractX509(const std::string& key, Jwks::Pubkey* jwk) {
   if (jwk->x509_ == nullptr) {
     return Status::JwksX509ParseError;
   }
-  jwk->evp_pkey_.reset(X509_get_pubkey(jwk->x509_.get()));
-  if (jwk->evp_pkey_ == nullptr) {
+  jwk->rsa_.reset(EVP_PKEY_get1_RSA(X509_get_pubkey(jwk->x509_.get())));
```
Comment (qiwzhang) [inline src/jwks.cc]: does EVP_PKEY_get1_RSA() handle nullptr input?  X509_get_pubkey may return nullptr.

File: src/verify.cc [hunk 2]
```diff
@@ -35,16 +35,21 @@ inline const uint8_t* castToUChar(const absl::string_view& str) {
...
+bool verifySignatureRSA(RSA* key, const EVP_MD* md, const uint8_t* signature,
+                        size_t signature_len, const uint8_t* signed_data,
+                        size_t signed_data_len) {
   if (key == nullptr || md == nullptr || signature == nullptr ||
       signed_data == nullptr) {
     return false;
   }
-  bssl::UniquePtr<EVP_MD_CTX> md_ctx(EVP_MD_CTX_create());
+  bssl::UniquePtr<EVP_PKEY> evp_pkey(EVP_PKEY_new());
+  if (EVP_PKEY_set1_RSA(evp_pkey.get(), key) != 1) {
```
Comment (qiwzhang) [inline src/verify.cc]: How slow is this EVP_PKey_set1_RSA call?  We are doing this for every JWT verification.  

In jwks parsing: (in envoy jwt_authn filter, this is done once for each jwks)
* RSA format:  generate RSA directly,
* pkcs and x509,   generate Pkey, and convert to RSA, 

in every verify() call,  we convert RSA back to Pkey.

I feel for better performance, we should keep Pkey format in jwks parsing.

Comment (alexcope) [reply src/verify.cc]: I did a quick perf profile by running a jwks_rsa test 100 times in a loop.  The conversion took 0.002% of the run time, so I don't think the performance of the conversion is an issue.

File: src/jwks.cc (hunk 1 above)
Comment (alexcope) [reply src/jwks.cc]: It does not, good catch.

**Pull request merged**